src/dateutil/parser/chain_parser.py

```diff
@@ -0,0 +1,71 @@
+import datetime as dt
+
+
+class _FixedStrptime(object):
+    """Parses datetime from strings by using a format that will be fed to strptime"""
+    def __init__(self, format_):
+        self._format = format_
+
+    def __call__(self, dt_str):
+        return dt.datetime.strptime(dt_str, self._format)
+
+    def __repr__(self):
+        return "StrptimeParser(%r)" % (self._format)
+
+
+class ChainParser(object):
+    """Parser that is built from combining multiple parsers
+
+    All provided parsers should have a call method that takes a string as input
```

Comment:
It is probably worth an additional paragraph about why you would want to use this:

```
This is useful for situations where you know that your dates are in one of a
small number of formats, or where you want to specifically prioritize some
formats over others. For example, if you want to interpret a 6-digit number as
``YYMMDD``, then ``DDMMYY``, then fall back to `dateutil.parser.parse` for any
strings that don't match either of those, you could construct the following
``ChainParser``:

.. doctest:: chainparser

    >>> parser = ChainParser("%y%m%d",  "%d%m%y", dateutil.parser.parse)
    >>> print(parser("120101"))
    2012-01-01 00:00:00
    >>> print(parser("010199"))
    1999-01-01 00:00:00
```

If we're including the example, we should probably put it right before the other example rather than before explaining the inputs.

Reply:
I'm thinking on deleting the other example and just leave this one